tests/api.c

```diff
@@ -9605,15 +9761,89 @@ static void test_wolfSSL_DES_ecb_encrypt(void)
     int ret1 = 0;
     int ret2 = 0;
     wolfSSL_DES_ecb_encrypt(&output1,&back1,&key,DES_DECRYPT);
-    ret1 = memcmp((unsigned char *) back1,(unsigned char *) input1,sizeof(WOLFSSL_DES_cblock));
+    ret1 = XMEMCMP((unsigned char *) back1,(unsigned char *) input1,sizeof(WOLFSSL_DES_cblock));
     AssertIntEQ(ret1,0);
     wolfSSL_DES_ecb_encrypt(&output2,&back2,&key,DES_DECRYPT);
-    ret2 = memcmp((unsigned char *) back2,(unsigned char *) input2,sizeof(WOLFSSL_DES_cblock));
+    ret2 = XMEMCMP((unsigned char *) back2,(unsigned char *) input2,sizeof(WOLFSSL_DES_cblock));
     AssertIntEQ(ret2,0);
 
     printf(resultFmt, passed);
     #endif
 }
+
+static void test_wolfSSL_ASN1_TIME_adj(void)
+{
+#if defined(OPENSSL_EXTRA) && !defined(NO_ASN1_TIME) \
+&& !defined(USER_TIME) && !defined(TIME_OVERRIDES)
+
+    const int year = 365*24*60*60;
+    const int day = 24*60*60;
+    const int hour = 60*60;
+    const int min = 60;
+    const byte asn_utc_time = 0x17;
+    const byte asn_gen_time = 0x18;
+    WOLFSSL_ASN1_TIME *asn_time, *s;
+    int offset_day;
+    long offset_sec;
+    char date_str[20];
+
+    printf(testingFmt, "wolfSSL_ASN1_TIME_adj()");
+
+    s = (WOLFSSL_ASN1_TIME*)XMALLOC(sizeof(WOLFSSL_ASN1_TIME), NULL,
+                                    DYNAMIC_TYPE_OPENSSL);
+    /* UTC notation test */
+    /* 2000/2/15 20:30:00 */
+    time_t t = (time_t)30 * year + 45 * day + 20 * hour + 30 * min + 7 * day;
+    offset_day = 7;
+    offset_sec = 45 * min;
+    /* offset_sec = -45 * min;*/
+    asn_time = wolfSSL_ASN1_TIME_adj(s, t, offset_day, offset_sec);
```

Comment:
Please also add a test case for `wolfSSL_ASN1_TIME_adj(NULL` so we can verify the creation of it in the function call is properly allocated and free'd.

Reply:
I added test case which called wolfSSL_ASN1_TIME_adj(NULL,....).